core/src/main/java/org/apache/iceberg/OverwriteData.java

```diff
@@ -19,15 +19,23 @@
 
 package org.apache.iceberg;
 
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
+import java.util.ArrayList;
 import java.util.List;
 import org.apache.iceberg.exceptions.ValidationException;
 import org.apache.iceberg.expressions.Evaluator;
 import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.InclusiveMetricsEvaluator;
 import org.apache.iceberg.expressions.Projections;
 import org.apache.iceberg.expressions.StrictMetricsEvaluator;
 
 public class OverwriteData extends MergingSnapshotProducer<OverwriteFiles> implements OverwriteFiles {
-  private boolean validateAddedFiles = false;
+  private boolean overwriteByRowFilter;
+  private boolean validateAddedFilesMatchOverwriteFilter;
+  private boolean deleteFilesByPath;
+  private Long readSnapshotId;
+  private Expression conflictDetectionFilter;
```

Comment:
I'd prefer to add explicit defaults for readability. Is there a reason to remove them?